crates/full-node/sov-sequencer/src/preferred/rate_limiter/mod.rs

```diff
@@ -91,13 +91,16 @@ const TTL_MULTIPLIER: u64 = 20;
 
 fn clculate_limits<S: Spec>(
     limits: Limits,
+    max_requests_per_second: u64,
     batch_execution_time_limit_millis: u64,
     max_batch_size_bytes: usize,
 ) -> RateLimiterConfig<S> {
     let max_gas = comfortable_gas_limit::<S>();
 
     let max_resources_per_batch = Resource {
-        req_counter: limits.max_requests_per_batch,
+        req_counter: max_requests_per_second
+            .checked_mul(batch_execution_time_limit_millis)
+            .expect("Unable to covert max_requests_per_second to req_counter"),
```

Comment:
Maybe this could be more explicit
```suggestion
            .expect("Overflow converting max_requests_per_second to max requests per batch"),
```

Reply:
fixed